subprojects/docs/src/docs/userguide/authoring-builds/multi-project/declaring_dependencies_between_subprojects.adoc

```diff
@@ -0,0 +1,134 @@
+====
+
+Shared build logic is extracted into a _convention plugin_ that is applied in the subprojects' build scripts that also define project dependencies.
+A project dependency is a special form of an execution dependency.
+It causes the other project to be built first and adds the jar with the classes of the other project to the classpath.
+It also adds the dependencies of the other project to the classpath.
+You can trigger a `gradle :api:compile`. First the `shared` project is built and then the `api` project is built.
+Project dependencies enable partial multi-project builds.
+
+
+[[sec:depending_on_output_of_another_project]]
+== Depending on output produced by another project
+
+<<#sec:project_jar_dependencies,Project dependencies>> model dependencies between modules.
+Effectively, you are saying that you depend on the main output of another project. In a Java-based project that's usually a JAR file.
+
+Sometimes you may want to depend on an output produced by another task.
+In turn, you'll want to make sure that the task is executed beforehand to produce that very output.
+Declaring a task dependency from one project to another is a poor way to model this kind of relationship and introduces unnecessary coupling.
+The recommended way to model such a dependency is to produce the output, mark it as an "outgoing" artifact or add it to the output of the `main` source set which you can depend on in the consuming project.
+
+Let's say you are working in a multi-project build with the two subprojects `producer` and `consumer`.
+The subproject `producer` defines a task named `buildInfo` that generates a properties file containing build information e.g. the project version.
+You can then map the task provider to its output file and Gradle will automatically establish a task dependency.
+
+.Task generating a property file containing build information
+====
+include::sample[dir="snippets/multiproject/dependencies-outgoingArtifact/groovy/producer",files="build.gradle[]"]
+include::sample[dir="snippets/multiproject/dependencies-outgoingArtifact/kotlin/producer",files="build.gradle.kts[]"]
+====
+
+The consuming project is supposed to be able to read the properties file at runtime.
+Declaring a project dependency on the producing project takes care of creating the properties beforehand and making it available to the runtime classpath.
+
+.Declaring a project dependency on the project producing the properties file
+====
+include::sample[dir="snippets/multiproject/dependencies-outgoingArtifact/groovy/consumer",files="build.gradle[tags=producer-project-dependency]"]
+include::sample[dir="snippets/multiproject/dependencies-outgoingArtifact/kotlin/consumer",files="build.gradle.kts[tags=producer-project-dependency]"]
+====
+
+In the example above, the consumer now declares a dependency on the outputs of the `producer` project.
+
+
+== Depending on other artifacts between projects
```

Comment:
```suggestion
```

I think we should drop this headline now and instead adjust the one above. Both are the same just in other words (effectively, the artefacts are the outputs).